src/components/FileInput/index.js

```diff
@@ -1,17 +1,110 @@
-import React from "react";
+import React, { useState } from "react";
 import { useFormikContext } from "formik";
+import ReactCrop from "react-image-crop";
+import "react-image-crop/dist/ReactCrop.css";
+
 import FileInputBase from "../UI/FileInputBase";
+import Modal from "../UI/Modal";
 
 const FileInput = ({ name = "", title = "add avatar" }) => {
   const { setFieldValue } = useFormikContext();
+  const [imageRef, setImageRef] = useState(null);
+  const [isModal, setIsModal] = useState(false);
+  const [src, setSrc] = useState(null);
+  const [fileName, setFileName] = useState("");
+
+  const [crop, setCrop] = useState({
+    unit: "px",
+    aspect: 1,
+    width: 300,
+    height: 300,
+    y: 50,
+    x: 100,
+  });
+
+  const closeModal = () => setIsModal(false);
+  const openModal = () => setIsModal(true);
+  const onImageLoaded = (image) => {
+    setImageRef(image);
+  };
+
+  const onCropComplete = () => {
+    makeClientCrop(crop);
+  };
+
+  const onCropChange = (crop) => {
+    setCrop(crop);
+  };
+
+  const loadFile = (event) => {
+    const file = event.currentTarget.files[0];
+    openModal();
+    setFileName(file.name);
+    setFieldValue(name, file);
+    setSrc(URL.createObjectURL(file));
```

Comment:
```suggestion
    setFileName(file.name);
    setFieldValue(name, file);
    setSrc(URL.createObjectURL(file));
```
у тебя будет 1(модальное) + 3 рендера, это вызовет дополнительную нагрузку на компонент, просядет performance

Reply:
убрал отсюда setFieldValue